tools/coverage/jacoco-common.gradle

```diff
@@ -0,0 +1,111 @@
+apply plugin: 'jacoco'
+
+jacoco {
+    toolVersion = "$versions.jacoco"
+}
+
+tasks.withType(Test) {
+    jacoco.includeNoLocationClasses = true
+}
+
+private static void addPathIfExists(List<String> paths,
+                                    String directoryPath) {
+    if (new File(directoryPath).exists())
+        paths.add(directoryPath)
+}
+
+private static void addFileTreeIfExists(Project project, List<FileTree> fileTrees,
+                                 String directoryPath, List<String> excludeFilter = []) {
+    if (new File(directoryPath).exists())
+        fileTrees.add(project.fileTree(dir: directoryPath, excludes: excludeFilter))
+}
+
+static List<String> findSourceDirectories(Project project, String flavor = "", String buildType = "debug") {
+
+    final sourcePaths = []
+
+    // Main
+    addPathIfExists(sourcePaths, "${project.projectDir}/src/main/java")
+    addPathIfExists(sourcePaths, "${project.projectDir}/src/main/kotlin")
+
+    // Flavor
+    if (!flavor.isEmpty()) {
+        addPathIfExists(sourcePaths, "${project.projectDir}/src/$flavor/java")
+        addPathIfExists(sourcePaths, "${project.projectDir}/src/$flavor/kotlin")
+    }
+
+    // BuildType
+    if (!buildType.isEmpty()) {
+        addPathIfExists(sourcePaths, "${project.projectDir}/src/$buildType/java")
+        addPathIfExists(sourcePaths, "${project.projectDir}/src/$buildType/kotlin")
+    }
+    return sourcePaths
+}
+
+static List<ConfigurableFileTree> findClassDirectories(Project project, String flavor = "", String buildType = "debug") {
+
+    final filter = [
+            java  : ['**/R.class', '**/R$*.class', '**/BuildConfig.*'],
+            kotlin: []
+    ]
+
+    final classPaths = []
+
+    // BuildType
+    if (!buildType.isEmpty()) {
+        addFileTreeIfExists(project, classPaths, "${project.buildDir}/intermediates/classes/$buildType", filter.java)
+        addFileTreeIfExists(project, classPaths, "${project.buildDir}/intermediates/javac/$buildType/compile${buildType.capitalize()}JavaWithJavac", filter.java)
+        addFileTreeIfExists(project, classPaths, "${project.buildDir}/tmp/kotlin-classes/$buildType", filter.kotlin)
+    }
+
+    // Flavor
+    if (!flavor.isEmpty()) {
+        addFileTreeIfExists(project, classPaths, "${project.buildDir}/intermediates/classes/$flavor/$buildType", filter.java)
+        addFileTreeIfExists(project, classPaths, "${project.buildDir}/intermediates/javac/$flavor${buildType.capitalize()}/compile${flavor.capitalize()}${buildType.capitalize()}JavaWithJavac", filter.java)

```

Comment:
Can this capitalize logic be extracted into a separated method? This will make the code more readable.